jwst/extract_1d/soss_extract/soss_extract.py

```diff
@@ -753,71 +801,96 @@ def decontaminate_image(scidata_bkg, tracemodels, subarray):
-                       mask_rebuild, order, wave_grid, valid_cols, save_tiktests=False, tikfac_log_range=None):
+def _model_single_order(data_order, err_order, ref_file_args, mask_fit,
+                        mask_rebuild, order, wave_grid, valid_cols,
+                        tikfac_log_range, save_tiktests=False):
+    """
+    Extract an output spectrum for a single spectral order using the ATOCA
+    algorithm, testing a range of Tikhonov factors.
+    The Tikhonov factor is derived in two stages: first, ten factors are tested
+    spanning tikfac_log_range, and then a further 20 factors are tested across
+    2 orders of magnitude in each direction around the best factor from the first
+    stage.
+    The best-fitting model and spectrum are reconstructed using the best-fit Tikhonov factor
+    and respecting mask_rebuild.
+
+    Parameters
+    ----------
+    data_order : np.array
+        The 2D data array for the spectral order to be extracted.
+    err_order : np.array
+        The 2D error array for the spectral order to be extracted.
+    ref_file_args : list
+        The reference file arguments used by the ExtractionEngine.
+    mask_fit : np.array
+        Mask determining the aperture used for extraction. This typically includes
+        detector bad pixels and any pixels that are not part of the trace
+    mask_rebuild : np.array
+        Mask determining the aperture used for rebuilding the trace. This typically includes
+        only pixels that do not belong to either spectral trace, i.e., regions of the detector
+        where no real data could exist.
+    order : int
+        The spectral order to be extracted.
+    wave_grid : np.array
+        The wavelength grid used to model the data.
+    valid_cols : np.array
+        The columns of the detector that are valid for extraction.
+    tikfac_log_range : list
+        The range of Tikhonov factors to test, in log space.
+    save_tiktests : bool, optional.
+        If True, save the intermediate models and spectra for each Tikhonov factor tested.
+    
+    Returns
+    -------
+    model : np.array
+        Model derived from the best Tikhonov factor, same shape as data_order.
+    spec_list : list of SpecModel
+        If save_tiktests is True, returns a list of the model spectra for each Tikhonov factor tested,
+        with the best-fitting spectrum last in the list.
+        If save_tiktests is False, returns a one-element list with the best-fitting spectrum.
+    
+    Notes
+    -----
+    The last spectrum in the list of SpecModels lacks the "chi2", "chi2_soft_l1", "chi2_cauchy", and "reg"
+    attributes, as these are only calculated for the intermediate models. The last spectrum is not
+    necessarily identical to any of the spectra in the list, as it is reconstructed according to
+    mask_rebuild instead of fit respecting mask_fit.
+
+    # TODO are all of these behaviors for the last spec in the list the desired ones?
```

Comment:
still wondering about this, specifically with respect to not treating the "best" / reconstructed models any differently than the other test models